### ISSUE TITLE
Logging failed when logging an object with resource type field

### ISSUE THREAD

**Issue by char101** [itsgoingd/clockwork#219]
`json_encode` will fail with `Type is not supported` error when logging an object containing resource type field, e.g. a database connection. This causes `Request::$log` to be set as `false` and written as `NULL` in the database, which will then fail when called in https://github.com/itsgoingd/clockwork/blob/master/Clockwork/Request/Request.php#L205.

**Comment by itsgoingd** [comment]
Hey, I have an idea how to improve this (using `JSON_PARTIAL_OUTPUT_ON_ERROR`), will work on it for the next release.

**Comment by char101** [comment]
Hi,

Using that flag does not work, it only outputs the class name and and empty json object `{ }`. What is required is an object serializer that can handle resource type and cyclic reference. Clockwork already has such function unfortunately it does not handle cyclic reference.

**Comment by itsgoingd** [comment]
True, it won't help in this case. I was originally working on a more complete serializer, but I couldn't get it to work as well as I wanted to. I might have another try on that for the next release.
